### DISCUSSION TITLE
Does the vpk tool prevent overwriting an existing version?

### DISCUSSION THREAD

**Discussion by bzuillsmith** [velopack/velopack.docs#9] [Q&A] [answered]
In our ci/cd pipeline, do we need to check if the version already exists before running the `vpk` commands or does the tool already do a check somewhere in the `download`/`pack`/`upload` steps?

**Comment by caesay** [comment, marked as answer]
By default, `vpk pack` will stop and ask you (y/n) if you want to overwrite an existing release on local disk. You can set non-interactive mode with `-x`, which will then not ask and default to No/fail to pack. If you wish for vpk to unconditionally overwrite the file, you can specify `-xy` instead to answer Yes to the question and continue to overwrite the local package. 

`vpk upload` will not re-upload any local assets if the MD5 checksum of the local file matches that of the remote file. If the checksums are different, then it will overwrite the remote file.

So assuming your CI is using `vpk download` to download the existing release (let's say it's `1.2.3`), if your CI is trying to build the same version `1.2.3`, it will fail by default with the `-x` argument.
